magik-language-server/src/main/java/nl/ramsolutions/sw/magik/languageserver/formatting/FormattingProvider.java

```diff
@@ -42,6 +41,49 @@ public List<TextEdit> provideFormatting(
     return textEdits.stream().map(Lsp4jConversion::textEditToLsp4j).toList();
   }
 
+  /**
+   * Provide formatting for text in range.
+   *
+   * @param magikFile Magik file.
+   * @param options Formatting options.
+   * @param range Range.
+   * @return {@link TextEdit}s.
+   * @throws IOException -
+   */
+  public List<TextEdit> provideRangeFormatting(
+      final MagikFile magikFile, final FormattingOptions options, final Range range) {
+    final List<org.eclipse.lsp4j.TextEdit> textEdits = this.provideFormatting(magikFile, options);
+    return textEdits.stream().filter(edit -> isEditInRange(edit, range)).toList();
+  }
+
+  /**
```

Comment:
You should use `Position.isBefore()`/`Position.isAfter()`/`Position.isBeforeRange()`/`Position.isAfterRange()` for this.